src/XLParser/ExcelFormulaGrammar.cs

```diff
@@ -251,32 +259,41 @@ public ExcelFormulaGrammar() : base(false)
             #region References
 
             Reference.Rule = ReferenceItem
-                | Reference + colon + Reference
-                | Reference + intersectop + Reference
-                | OpenParen + Union + CloseParen
+                | ReferenceFunctionCall
                 | OpenParen + Reference + PreferShiftHere() + CloseParen
                 | Prefix + ReferenceItem
-                | Prefix + UDFToken + Arguments + CloseParen
                 | DynamicDataExchange
                 ;
 
+            ReferenceFunctionCall.Rule =
+                  Reference + colon + Reference
+                | Reference + intersectop + Reference
+                | OpenParen + Union + CloseParen
+                | RefFunctionName + Arguments + CloseParen
+                //| ConditionalRefFunctionName + Arguments + CloseParen
+                ;
+
+            RefFunctionName.Rule = ExcelRefFunctionToken | ExcelConditionalRefFunctionToken;
+
             Union.Rule = MakePlusRule(Union, comma, Reference);
 
             ReferenceItem.Rule =
                 Cell
                 | NamedRange
-                | ReferenceFunction
                 | VRange
                 | HRange
                 | RefError
+                | UDFunctionCall
```

Comment:
If UDF function calls are handled as reference function calls, the two non terminals
```RefFunctionCall = RefFunctionName + Arguments + CloseParen``` and
```UDFunctionCall = UDFName + Arguments + CloseParen```
could both be either in the ```ReferenceFunctionCall.Rule``` or in the ```ReferenceItem.Rule```. Then they would be in the same level in the parse tree.